polaris/hub/client.py

```diff
@@ -333,13 +333,16 @@ def get_dataset(self, owner: Union[str, HubOwner], name: str) -> Dataset:
 
         return Dataset(**response)
 
-    def read_zarr_file(self, owner: Union[str, HubOwner], name: str, path: str) -> zarr.hierarchy.Group:
-        """Read a Zarr file from a Polaris dataset
+    def open_zarr_file(
+        self, owner: Union[str, HubOwner], name: str, path: str, mode: str
```

Comment:
Instead of the `str` type annotation, could we come up with a more specific type hint? From the [Zarr source code](https://github.com/zarr-developers/zarr-python/blob/a1fbedb18c1fc70f026c423fafca6d84ad88ce53/zarr/convenience.py#L49-L53), it seems Zarr supports a bit more than just `r` and `w`.

Maybe we can introduce a custom type in [`polaris/utils/types.py`](https://github.com/polaris-hub/polaris/blob/main/polaris/utils/types.py) For example: 
```python
IOMode: TypeAlias = Literal["r", "r+", "a", "w", "w-"]
```